src/SocketControllerExecutor.ts

```diff
@@ -216,14 +216,18 @@ export class SocketControllerExecutor {
         }
     }
 
-    private handleSuccessResult(result: any, action: ActionMetadata, socket: any) {
+    private handleSuccessResult(result: any, action: ActionMetadata, socket: any, clientCallback: Function) {
         if (result !== null && result !== undefined && action.emitOnSuccess) {
             const transformOptions = action.emitOnSuccess.classTransformOptions || this.classToPlainTransformOptions;
             let transformedResult = this.useClassTransformer && result instanceof Object ? classToPlain(result, transformOptions) : result;
             socket.emit(action.emitOnSuccess.value, transformedResult);
 
         } else if ((result === null || result === undefined) && action.emitOnSuccess && !action.skipEmitOnEmptyResult) {
             socket.emit(action.emitOnSuccess.value);
+        } else if (result !== null && result !== undefined && clientCallback) {
+            clientCallback(result);
+        } else if ((result === null || result === undefined) && clientCallback) {
+            clientCallback("received");
```

Comment:
Received is for situations where the function returns nothing but the client is expecting a callback. Any good ideas for this?